### DISCUSSION TITLE
DockerHub rate limit exceeded

### DISCUSSION THREAD

**Discussion by Anton-Shutik** [dflook/terraform-github-actions#330] [Q&A] [answered]
I have an issue with dockerhub rate limit, since have a lot of jobs running on my self-hosted runner. 

Is there any way I can specify my own pull-through ([AWS ECR for example](https://docs.aws.amazon.com/AmazonECR/latest/userguide/pull-through-cache.html)) in my `.yaml` file, so it will not pull the image from the dockerhub every time ?

```python
Pull down action image 'danielflook/terraform-github-actions@sha256:31f49b44966fdd5a6595ba90b8ca2d0226199f82c8bc1826654b00b5d8e04fc3'
  /usr/local/bin/docker pull danielflook/terraform-github-actions@sha[2]
  Error response from daemon: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
  Warning: Docker pull failed with exit code 1, back off 2.56 seconds before retry.
  /usr/local/bin/docker pull danielflook/terraform-github-actions@sha256:31f49b44966fdd5a6595ba90b8ca2d0226199f82c8bc1826654b00b5d8e04fc3
  Error response from daemon: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
  Warning: Docker pull failed with exit code 1, back off 8.524 seconds before retry.
  /usr/local/bin/docker pull danielflook/terraform-github-actions@sha256:[3]
  Error response from daemon: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
Error: Docker pull failed with exit code 1
```

**Comment by dflook** [comment, marked as answer]
There is a published variant that uses ghcr, which doesn't seem to have dockerhubs low rate limit.

You can use it by adding '-gchr' to the action version.

e.g.
```yaml
jobs:
  build:
    steps:
    - uses: dflook/terraform-plan@v1.41.2-ghcr
```

**Reply by Anton-Shutik** [reply]
Thanks,

That said, there is no way to use my own container repo for that, right ?

**Reply by dflook** [reply]
Right, there is no way to specify your own container image, unfortunately.

**Reply by Anton-Shutik** [reply]
Looks like ghcr has [their rate limits as well](https://github.com/orgs/community/discussions/45253). 

Are there any plans to add an ability to use customer repo (ex. Private ECR backed by dockerhub ?) ?

**Reply by dflook** [reply]
I would if I could. Your only option is to fork, see [here](https://github.com/dflook/terraform-github-actions/blob/main/docs/custom_tools.md#using-a-pre-built-custom-image).
